sport-moment-nft/contracts/Golazos.cdc

```diff
@@ -585,26 +585,33 @@ pub contract Golazos: NonFungibleToken {
         ///
         pub fun name(): String {
             let editionData = Golazos.getEditionData(id: self.editionID)!
-            let fullName: String = Golazos.PlayData(id: editionData.playID).metadata["PlayerJerseyName"] ?? ""
+            let playerKnownName: String = Golazos.PlayData(id: editionData.playID).metadata["PlayerKnownName"] ?? ""
+            let playerFirstName: String = Golazos.PlayData(id: editionData.playID).metadata["PlayerFirstName"] ?? ""
+            let playerLastName: String = Golazos.PlayData(id: editionData.playID).metadata["PlayerLastName"] ?? ""
             let playType: String = Golazos.PlayData(id: editionData.playID).metadata["PlayType"] ?? ""
-            return fullName
-                .concat(" ")
-                .concat(playType)
+            var playerName = playerKnownName
+            if(playerName == ""){
+                playerName = playerFirstName.concat(" ").concat(playerLastName)
+            }
+            return playType.concat(" by ").concat(playerName)
         }
 
         /// get the description of an nft
         ///
         pub fun description(): String {
             let editionData = Golazos.getEditionData(id: self.editionID)!
-            let setName: String = Golazos.SetData(id: editionData.setID)!.name
-            let serialNumber: String = self.serialNumber.toString()
-            let seriesNumber: String = editionData.seriesID.toString()
-            return "A series "
-                .concat(seriesNumber)
-                .concat(" ")
-                .concat(setName)
-                .concat(" moment with serial number ")
-                .concat(serialNumber)
+            let metadata = Golazos.PlayData(id: editionData.playID).metadata
+            let matchHomeTeam: String = metadata["MatchHomeTeam"] ?? ""
+            let matchAwayTeam: String = metadata["MatchAwayTeam"] ?? ""
+            let matchHomeScore: String = metadata["MatchHomeScore"] ?? ""
+            let matchAwayScore: String = metadata["MatchAwayScore"] ?? ""
+            let matchDay: String = metadata["MatchDay"] ?? ""
+            let matchSeason: String = metadata["MatchSeason"] ?? ""
+
+            return "LaLiga Golazos Moment from ".concat(matchHomeTeam)
+            .concat(" x ").concat(matchAwayTeam).concat(" (").concat(matchHomeScore)
+            .concat("-").concat(matchAwayScore).concat(") on Matchday ")
```

Comment:
Nice! What would end up displaying if the scores were 0 or null?

Reply:
should display 0-0 which is a valid scoreline. Should not have a value of null tho

Reply:
0-0 sounds good 👍  is it possible to be null at all? Like if it was a special NFT or something? (although we'd probably want to use a different contract if that was the case

Reply:
yea, also. since it's a {String:String} type, it won't accept a null value

Reply:
oh nice, 👍